src/bucket.js

```diff
@@ -468,6 +470,141 @@ Bucket.prototype.createChannel = function(id, config, options, callback) {
+ *     listed event types. If empty, sent notifications for all event types.
+ * @property {string} [objectNamePrefix] If present, only apply this
+ *     notification configuration to object names that begin with this prefix.
+ * @property {string} [payloadFormat] The desired content of the Payload.
+ *     Defaults to `JSON_API_V1`.
+ *
+ *     Acceptable values are:
+ *     - `JSON_API_V1`
+ *
+ *     - `NONE`
+ * @property {boolean} [options.userProject] If this bucket has `requesterPays`
+ *     functionality enabled (see {@link Bucket#enableRequesterPays}),
+ *     set this value to the project which should be billed for this operation.
+ */
+/**
+ * @typedef {array} CreateNotificationResponse
+ * @property {Notification} 0 The new {@link Notification}
+ * @property {object} 1 The full API response.
+ */
+/**
+ * @callback CreateNotificationCallback
+ * @param {?Error} err Request error, if any.
+ * @param {Notification} notification The new {@link Notification}
+ * @param {object} apiResponse The full API response.
+ */
+/**
+ * Creates a notification subscription for the bucket.
+ *
+ * @see [Notifications: insert]{@link https://cloud.google.com/storage/docs/json_api/v1/notifications/insert}
+ *
+ * @param {Topic|string} topic The Cloud PubSub topic to which this
+ *     subscription publishes. If the project ID is omitted, the current project
+ *     ID will be used.
+ *
+ *     Acceptable formats are:
+ *     - `projects/grape-spaceship-123/topics/my-topic`
+ *
+ *     - `my-topic`
+ * @param {CreateNotificationRequest} [options] Metadata to set for the
+ *     notification.
+ * @param {CreateNotificationCallback} [callback] Callback function.
+ * @returns {Promise<CreateNotificationResponse>}
+ * @throws {Error} If a valid topic is not provided.
+ * @see Notification#create
+ *
+ * @example
+ * var storage = require('@google-cloud/storage')();
+ * var myBucket = storage.bucket('my-bucket');
+ *
+ * var callback = function(err, notification, apiResponse) {
+ *   if (!err) {
+ *     // The notification was created successfully.
+ *   }
+ * };
+ *
+ * myBucket.createNotification('my-topic', callback);
+ *
+ * //-
+ * // Configure the nofiication by providing Notification metadata.
+ * //-
+ * var metadata = {
+ *   objectNamePrefix: 'prefix-'
+ * };
+ *
+ * myBucket.createNotification('my-topic', metadata, callback);
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * myBucket.createNotification('my-topic').then(function(data) {
+ *   var notification = data[0];
+ *   var apiResponse = data[1];
+ * });
+ *
+ * @example <caption>include:samples/notifications.js</caption>
+ * region_tag:storage_create_notification
+ * Another example:
+ */
+Bucket.prototype.createNotification = function(topic, options, callback) {
+  var self = this;
+
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  if (is.object(topic)) {
```

Comment:
We have the `util.isCustomType` method we can use here.

Reply:
Done